Compilation error in JFlexRunner.scala when building with JFlex 1.5+

When building with JFlex 1.5 or newer, it suddenly can't seem to find JFlex anymore, and the [following error](https://github.com/EwoutH/NetLogo/runs/4164358300?check_suite_focus=true#step:6:197) presents itself on compilation. The error is identical for JFlex 1.5.1, 1.6.1 and 1.8.2.
```
Error:  /home/runner/work/NetLogo/NetLogo/project/JFlexRunner.scala:30:5: not found: value JFlex
Error:      JFlex.Main.main(Array("--quiet", "-d", cacheDirectory.getPath, (autogenFolder / "flex" / (kind + ".flex")).asFile.getAbsolutePath.toString))
Error:      ^
Error:  one error found
Error:  (Compile / compileIncremental) Compilation failed
Project loading failed: (r)etry, (q)uit, (l)ast, or (i)gnore? 
Error: Process completed with exit code 1.
```
The error is located in [project/JFlexRunner.scala](https://github.com/NetLogo/NetLogo/blob/6.2.1/project/JFlexRunner.scala#L30). JFlex can be updated in [project/plugins.sbt](https://github.com/NetLogo/NetLogo/blob/6.2.1/project/plugins.sbt#L28).

Part of #1968.

--- a/autogen/flex/ImportLexer.flex
+++ b/autogen/flex/ImportLexer.flex
@@ -8,9 +8,6 @@ package org.nlogo.agent;
 
 %%
 
-/* this option decreases code size; see JFlex documentation */
-%switch
-
 %unicode
 
 %yylexthrow{

--- a/project/JFlexRunner.scala
+++ b/project/JFlexRunner.scala
@@ -27,7 +27,8 @@ object JFlexRunner {
   // this used to be broken into two tasks, but jflex doesnt seem to be threadsafe
   // so we have to run them serially, which means we have to generate both files each time. -JC 6/8/10
   def flex(log: String => Unit, autogenFolder: File, cacheDirectory: File, srcRoot: File, ppackage: String, kind: String): File = {
-    JFlex.Main.main(Array("--quiet", "-d", cacheDirectory.getPath, (autogenFolder / "flex" / (kind + ".flex")).asFile.getAbsolutePath.toString))
+    // when we move to JFlex 1.6+, --noinputstreamctor can be removed
+    jflex.Main.main(Array("--quiet", "--noinputstreamctor", "-d", cacheDirectory.getPath, (autogenFolder / "flex" / (kind + ".flex")).asFile.getAbsolutePath.toString))
     val destination = (srcRoot / "org" / "nlogo" / ppackage / (kind + ".java")).getAbsoluteFile
     log("creating " + srcRoot + "/org/nlogo/" + ppackage + "/" + kind + ".java")
     IO.write(destination,

--- a/project/NetLogoDocsTest.scala
+++ b/project/NetLogoDocsTest.scala
@@ -26,16 +26,16 @@ object NetLogoDocsTest {
   def apply(docsRoot: File): Map[String, Seq[String]] = {
     val testedLinks = new JHashMap[String, JBoolean]
 
-    val fileToLines = docsRoot.listFiles filter (_.getName.endsWith(".html")) map { file: File =>
+    val fileToLines = docsRoot.listFiles.filter(_.getName.endsWith(".html")).map{ file: File =>
       import scala.collection.JavaConverters._
       file.getName -> Files.readAllLines(file.toPath).asScala
-    } toMap;
+    }.toMap
 
     fileToLines mapValues { lines =>
-      val urls = lines flatMap { line =>
+      val urls = lines.flatMap{ line =>
         (linkRegex.findAllMatchIn(line) ++ imgRegex.findAllMatchIn(line))
           .map(_.group(1)).toSeq
-      } distinct;
+      }.distinct
 
       urls filterNot {
         case hrefRegex(null, null, anchor) =>

--- a/project/build.sbt
+++ b/project/build.sbt
@@ -1,2 +1,1 @@
-lazy val root = project.in(file("."))
-  .settings(scalacOptions += "-deprecation")
+scalacOptions ++= Seq("-feature", "-deprecation")

--- a/project/plugins.sbt
+++ b/project/plugins.sbt
@@ -25,7 +25,7 @@ addSbtPlugin("com.timushev.sbt"   %  "sbt-updates"                     % "0.6.0"
 
 libraryDependencies ++= Seq(
   "com.github.spullara.mustache.java" % "compiler"              % "0.9.5"
-, "de.jflex"                          % "jflex"                 % "1.4.3"
+, "de.jflex"                          % "jflex"                 % "1.5.1"
 , "classycle"                         % "classycle"             % "1.4.2" from
   "https://s3.amazonaws.com/ccl-artifacts/classycle-1.4.2.jar"
 , "com.github.spullara.mustache.java" % "scala-extensions-2.10" % "0.9.5"

